### PR TITLE
[feat] Impose units in performance reference tuple

### LINKED ISSUE #1147: Impose units in references
Currently, units are optional, in order to allow backward compatibility. I suggest to enforce them for ReFrame 3.0. If somebody does not want a unit, she should use an empty string or `None` explicitly.

### PR DESCRIPTION
**Pull request by teojgo**
* Fix checks that don't contain units in performance reference tuples.

* Add unit test.

* Documentation fixes.

Fixes #1147

### CODE REVIEW

File: cscs-checks/mch/fieldextra_check.py [hunk 1]
```diff
@@ -84,6 +84,6 @@ def __init__(self):
         }
         self.reference = {
             'kesch': {
-                'perf': (420., None, 0.10)
+                'perf': (420., None, 0.10, None)
```
[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/mch/fieldextra_check.py]: @kraushm What are the units here? We need to merge this asap.

File: cscs-checks/mch/g2g_meteoswiss_check.py [hunk 2]
```diff
@@ -46,6 +46,6 @@ def __init__(self, g2g):
                                      self.stdout, 'time', float)
         }
         self.reference = {
-            'kesch:cn': {'time': (3.461, None, 0.2)}
+            'kesch:cn': {'time': (3.461, None, 0.2, None)}
```
[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/mch/g2g_meteoswiss_check.py]: @ajocksch What are the units here? We need to merge this asap.

@teojgo Can you please update the maintainers for this. They seem completely random... Put AJ and LM.

File: cscs-checks/mch/fieldextra_check.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (teojgo) [reply cscs-checks/mch/fieldextra_check.py]: @vkarak they seem to be seconds, I looked a little bit at the output formatting of the code.

File: cscs-checks/mch/fieldextra_check.py [hunk 3]
```diff
@@ -84,6 +84,6 @@ def __init__(self):
         }
         self.reference = {
             'kesch': {
-                'perf': (420., None, 0.10, None)
+                'perf': (420., None, 0.10, 's')
```
Comment (vkarak) [inline cscs-checks/mch/fieldextra_check.py]: The name of the variable should be `time` then.

**Comment by vkarak** [comment]
@jenkins-cscs retry none

**Pull request merged**